clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4845,7 +4904,7 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     }
 
     C.addCommand(std::make_unique<Command>(
-        JA, *this, ResponseFileSupport::AtFileUTF8(), D.getClangProgramPath(),
+        Job, *this, ResponseFileSupport::AtFileUTF8(), D.getClangProgramPath(),
```

Comment:
Why is this different from the others?

Reply:
This is intentional. The command creates should be associated with the original job requested, not the overwritten Job. The difference is simply for Depscan job, the command created here is associated with Depscan job, not a compile job.